cmd/grafana-app-sdk/generate.go

```diff
@@ -194,3 +243,62 @@ func generateCRDs(parser *themagen.CustomKindParser, genPath string, encoding st
 	}
 	return files, nil
 }
+
+func generateKindsCue(modFS fs.FS, cfg kindGenConfig, selectors ...string) (codejen.Files, error) {
+	parser, err := cuekind.NewParser()
+	if err != nil {
+		return nil, err
+	}
+	generator, err := codegen.NewGenerator[codegen.Kind](parser, modFS)
+	if err != nil {
+		return nil, err
+	}
+	// Resource
+	resourceFiles, err := generator.FilteredGenerate(cuekind.ResourceGenerator(true), func(kind codegen.Kind) bool {
+		return kind.Properties().APIResource != nil
+	}, selectors...)
+	if err != nil {
+		return nil, err
+	}
+	for i, f := range resourceFiles {
+		resourceFiles[i].RelativePath = filepath.Join(filepath.Join(cfg.GoGenBasePath, targetResource), f.RelativePath)
+	}
+	// Model
+	modelFiles, err := generator.FilteredGenerate(cuekind.ModelsGenerator(true), func(kind codegen.Kind) bool {
+		return kind.Properties().APIResource == nil
+	}, selectors...)
+	if err != nil {
+		return nil, err
+	}
+	for i, f := range modelFiles {
+		modelFiles[i].RelativePath = filepath.Join(filepath.Join(cfg.GoGenBasePath, targetModel+"s"), f.RelativePath)
+	}
+	// TypeScript
+	tsFiles, err := generator.Generate(cuekind.TypeScriptModelsGenerator(true), selectors...)
+	if err != nil {
+		return nil, err
+	}
+	for i, f := range tsFiles {
+		tsFiles[i].RelativePath = filepath.Join(cfg.TSGenBasePath, f.RelativePath)
+	}
+	// CRD
+	encFunc := json.Marshal
+	if cfg.CRDEncoding == "yaml" {
+		encFunc = yaml.Marshal
+	}
+	crdFiles, err := generator.FilteredGenerate(cuekind.CRDGenerator(encFunc, cfg.CRDEncoding), func(kind codegen.Kind) bool {
+		return kind.Properties().APIResource != nil
+	}, selectors...)
+	if err != nil {
+		return nil, err
+	}
+	for i, f := range crdFiles {
+		crdFiles[i].RelativePath = filepath.Join(cfg.CRDPath, f.RelativePath)
+	}
+
+	allFiles := append(make(codejen.Files, 0), resourceFiles...)
+	allFiles = append(allFiles, modelFiles...)
+	allFiles = append(allFiles, tsFiles...)
+	allFiles = append(allFiles, tsFiles...)
```

Comment:
Looks like instead of appending CRD files this is appending TS files two times.
```suggestion
	allFiles = append(allFiles, crdFiles...)
```